openrndr-core/src/main/kotlin/org/openrndr/shape/ShapeOperations.kt

```diff
@@ -606,4 +606,38 @@ fun split(shape: Shape, line: LineSegment): Pair<Shape, Shape> {
     val leftShape = difference(shape, leftContour)
     val rightShape = difference(shape, rightContour)
     return Pair(leftShape, rightShape)
+}
+
+/**
+ * Splits a ShapeContour with another ShapeContour.
+ * If there is no intersection it returns the original contour.
+ */
+fun split(from: ShapeContour, cutter: ShapeContour): List<ShapeContour> {
+    if (from.empty || cutter.empty)
```

Comment:
I think when the cutter is empty this code should return `listOf(from)`

Reply:
Agree. Done :)